pytket/pytket/backends/backendinfo.py

```diff
@@ -13,13 +13,127 @@
 # limitations under the License.
 
 """ BackendInfo class: additional information on Backends """
+
+from ast import literal_eval
 from dataclasses import dataclass, field, asdict
 from typing import Any, Dict, List, Optional, Set, cast, Tuple, Union
 
 from pytket.routing import Architecture, FullyConnected  # type: ignore
 from pytket.circuit import Node, OpType  # type: ignore
 
 
+def _serialize_all_node_gate_errors(
+    d: Optional[Dict[Node, Dict[OpType, float]]]
+) -> Optional[Dict[str, Dict[int, float]]]:
+    if d is None:
+        return None
+    return {
+        str(n.to_list()): {ot.value: err for ot, err in errs.items()}
+        for n, errs in d.items()
+    }
+
+
+def _deserialize_all_node_gate_errors(
+    d: Optional[Dict[str, Dict[int, float]]]
+) -> Optional[Dict[Node, Dict[OpType, float]]]:
+    if d is None:
+        return None
+    return {
+        Node.from_list(literal_eval(n)): {
+            OpType(int(ot)): err for ot, err in errs.items()
+        }
+        for n, errs in d.items()
+    }
+
+
+def _serialize_all_edge_gate_errors(
+    d: Optional[Dict[Tuple[Node, Node], Dict[OpType, float]]]
+) -> Optional[Dict[str, Dict[int, float]]]:
+    if d is None:
+        return None
+    return {
+        str((n0.to_list(), n1.to_list())): {ot.value: err for ot, err in errs.items()}
+        for (n0, n1), errs in d.items()
+    }
+
+
+def _deserialize_all_edge_gate_errors(
+    d: Optional[Dict[str, Dict[int, float]]],
+) -> Optional[Dict[Tuple, Dict[OpType, float]]]:
+    if d is None:
+        return None
+    return {
+        tuple(map(Node.from_list, literal_eval(n))): {
+            OpType(int(ot)): err for ot, err in errs.items()
+        }
+        for n, errs in d.items()
+    }
+
+
+def _serialize_all_readout_errors(
+    d: Optional[Dict[Node, List[List[float]]]]
+) -> Optional[Dict[str, List[List[float]]]]:
+    if d is None:
+        return None
+    return {str(n.to_list()): errs for n, errs in d.items()}
+
+
+def _deserialize_all_readout_errors(
+    d: Optional[Dict[str, List[List[float]]]]
+) -> Optional[Dict[Node, List[List[float]]]]:
+    if d is None:
+        return None
+    return {Node.from_list(literal_eval(n)): errs for n, errs in d.items()}
+
+
+def _serialize_averaged_node_gate_errors(
+    d: Optional[Dict[Node, float]]
+) -> Optional[Dict[str, float]]:
+    if d is None:
+        return None
+    return {str(n.to_list()): err for n, err in d.items()}
+
+
+def _deserialize_averaged_node_gate_errors(
+    d: Optional[Dict[str, float]]
+) -> Optional[Dict[Node, float]]:
+    if d is None:
+        return None
+    return {Node.from_list(literal_eval(n)): err for n, err in d.items()}
```

Comment:
Looks like there's some code duplication that can be reduced with more generic type signatures (e.g. by using `Sequence` or type variables), e.g. these two functions and the two before contain essentially the same body